canals/component/component.py

```diff
@@ -15,6 +15,43 @@
 logger = logging.getLogger(__name__)
 
 
+# We ignore too-few-public-methods Pylint error as this is only meant to be
+# the definition of the Component interface.
+# A concrete Component will have more than method in any case.
+class Component(Protocol):  # pylint: disable=too-few-public-methods
+    """
+    Abstract interface of a Component.
+    This is only used by type checking tools.
+
+    If you want to create a new Component use the @component decorator.
```

Comment:
Does this mean that a `Component` class can be instantiated? Is there a way to raise an error when that happens?